scripts/cnn_feat_extraction/README.md

```diff
@@ -12,17 +12,57 @@
+
+*[Files]*
+- `model_dir` is the folder where caffe model definitions are stored.
+- `imageset_dir` is the imageset directory. It contains images or subdirectories. Do not include non-image files in the directory; while it will be rejected, it will mess up the imagecount.
+- `model_def` is the `.prototxt` file within `imageset_dir`.
+- `model_weights` is the `.caffemodel` file within `imageset_dir`.
+- `mean_image` is the `.npy` file within `imageset_dir`.
+- `out_dir` is the directory in which the extracted features will be stored. If it doesn't exist, it will be created.
+
+*[Params]*
+- `batch_size` is the batch size for the neural network. If on CPU, the recommended value is `1`. If on GPU, it can be increased to `16` or greater for a faster runtime.
+- `GPU_enable` defines whether to run caffe in CPU or GPU modes. `0` for CPU, `1` for GPU.
+- `layer_list` defines which layers to extract features from. Layer names are listed in the `.prototxt` file that comes with the pretrained model.
+- `class_list` defines how to deal with different imageset structures.
+    - `<all>` tells the script to look for images in all subdirectories. For this option, `qty_list` must also be defined as `<all>`.
+	- `<flat>` tells the script that there are no subdirectories; all of the images are in the top level of `imageset_dir`. `qty_list` can only be one value.
+	- `dir1,dir2,dir3` tells the script to look in the specified subdirectories for the images. `qty_list` must be the same length.
+- `qty_list` defines how many images to extract features from.
+    - `<all>` tells the script to extract from all images in the subdirectories specified.
+	- `10` tells the script to extract 10 images from `<flat>`, or the single subdirectory specified.
+	- `10,20,30` tells the script to extract 10, 20, and 30 images from the specified subdirectories, respectively.
+	- `10,all,30` - the `all` keyword can be mixed within the list to specify that all images in the respective subdirectory are too be extracted from.
+
+### 3. Feature Extraction
+
+Extract the CNN feature vectors, which will be saved to .csv files:
+
+`python feat_csv.py your_config_file.cfg`
 
-2. Create a config file to specify locations of the trained CNN (Caffe model), the image files, and other parameters.  You can use `example.cfg` as a starting point.
+The first column is a label of the image's subdirectory (if it exists) and the filename as `[subdir]-[filename]`. If the file was deemed not an image, the label will have `BAD` appended to it. 
 
-3. Extract the CNN feature vectors, which will be saved to .csv files:
+### 4. DEMUD usage
 
-`python feat_csv.py your_config_file.cfg` 
+The resulting .csv files can be read in by DEMUD with the `-v` or `--cnn` option.
 
-4. The resulting .csv files can be read in by DEMUD with the `-v` or `--cnn` option.
+### 5. Performance
 
```

Comment:
Suggest adding some more detail here (or omitting from public github).  If kept, explain 'b' (batch size), 'Para', and versions.

Reply:
This section doesn't really help anyone as of now, omitting.

Reply:
Maybe put in Evernote?